integration/cdi/src/main/java/org/apache/cxf/cdi/JAXRSCdiResourceExtension.java

```diff
@@ -74,16 +74,16 @@
             featureBeans.add((Bean< ? extends Feature >)event.getBean());
         } else if (CdiBusBean.CXF.equals(event.getBean().getName())
                 && Bus.class.isAssignableFrom(event.getBean().getBeanClass())) {
-            busBean = event.getBean();
+            hasBus = true;
         }
     }
 
     public void load(@Observes final AfterDeploymentValidation event, final BeanManager beanManager) {
+        // no need of creational context, it only works for app scoped instances anyway
+        final Bean<?> busBean = beanManager.resolve(beanManager.getBeans(CdiBusBean.CXF));
         bus = (Bus)beanManager.getReference(
```

Comment:
You are right, this is more rare to have these beans added programmatically since CXF is not able to read metadata from CDI but this is possible. That said not sure we should worry much yet on that case - can be another PR if a user hits it maybe.